helix-term/src/ui/picker/handlers.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    path::PathBuf,
+    sync::{atomic, Arc},
+    time::Duration,
+};
+
+use helix_event::AsyncHook;
+use tokio::time::Instant;
+
+use crate::ui::overlay::Overlay;
+
+use super::{CachedPreview, DynQueryCallback, Picker};
+
+pub(super) struct PreviewHighlightHandler<T: 'static + Send + Sync, D: 'static + Send + Sync> {
+    trigger: Option<Arc<PathBuf>>,
+    phantom_data: std::marker::PhantomData<(T, D)>,
+}
+
+impl<T: 'static + Send + Sync, D: 'static + Send + Sync> Default for PreviewHighlightHandler<T, D> {
+    fn default() -> Self {
+        Self {
+            trigger: None,
+            phantom_data: Default::default(),
+        }
+    }
+}
+
+impl<T: 'static + Send + Sync, D: 'static + Send + Sync> AsyncHook
+    for PreviewHighlightHandler<T, D>
+{
+    type Event = Arc<PathBuf>;
+
+    fn handle_event(
+        &mut self,
+        path: Self::Event,
+        timeout: Option<tokio::time::Instant>,
+    ) -> Option<tokio::time::Instant> {
+        if self
+            .trigger
+            .as_ref()
+            .is_some_and(|trigger| trigger == &path)
```

Comment:
A small performance hack:

Right now you always create a new arc so this won't work but we don't actually want to do that we want to reuse the existing Arc. 

In that case two paths being equal would mean they are the exact same location in memory. So you can use Arc::ptr_eq which is faster

Reply:
Ah actually reading trough the sourcecode a little more carefully it seems like we du actually create a new PathNuf (and therefore Arc) everytime we render (by calling file_func). So this doesn't actually work correctly. 

I do wonder if we could avoid that but let's not worry about that for now and just revert the check to be `==` again

Reply:
Sorry about that